src/SourceGenerators/Uno.UI.SourceGenerators/RemoteControl/RemoteControlGenerator.cs

```diff
@@ -40,31 +50,29 @@ public override void Execute(SourceGeneratorContext context)
 				BuildEndPointAttribute(context, sb);
 				BuildSearchPaths(context, sb);
 
-				context.AddCompilationUnit("RemoteControl", sb.ToString());
+				context.AddSource("RemoteControl", sb.ToString());
 			}
 		}
 
-		private static bool IsRemoteControlClientInstalled(SourceGeneratorContext context)
+		private static bool IsRemoteControlClientInstalled(GeneratorExecutionContext context)
 			=> context.Compilation.GetTypeByMetadataName("Uno.UI.RemoteControl.RemoteControlClient") != null;
 
-		private static void BuildSearchPaths(SourceGeneratorContext context, IndentedStringBuilder sb)
+		private static void BuildSearchPaths(GeneratorExecutionContext context, IndentedStringBuilder sb)
 		{
-			var projectInstance = context.GetProjectInstance();
-
 			sb.AppendLineInvariant($"[assembly: global::Uno.UI.RemoteControl.ProjectConfigurationAttribute(");
-			sb.AppendLineInvariant($"@\"{projectInstance.FullPath}\",\n");
+			sb.AppendLineInvariant($"@\"{context.GetMSBuildPropertyValue("MSBuildProjectFullPath")}\",\n");
 
+			var msBuildProjectDirectory = context.GetMSBuildPropertyValue("MSBuildProjectDirectory");
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'msBuildProjectDirectory'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6270956)